analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -43,9 +44,77 @@ protected override ProgramState[] Process(SymbolicContext context, IInvocationOp
         {
             state = state.ResetFieldConstraints();
         }
-        return invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace))
-            ? ProcessStringIsNullOrEmpty(context, invocation)
-            : new[] { state };
+        if (invocation.TargetMethod.ContainingType.Is(KnownType.System_Linq_Enumerable))
+        {
+            return ProcessLinqEnumerable(context, invocation);
+        }
+        else if (invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace)))
+        {
+            return ProcessStringIsNullOrEmpty(context, invocation);
+        }
+        else
+        {
+            return new[] { state };
+        }
+    }
+
+    private static ProgramState[] ProcessLinqEnumerable(SymbolicContext context, IInvocationOperationWrapper invocation)
+    {
+        switch (invocation.TargetMethod.Name)
+        {
+            case "Append":
+            case nameof(Enumerable.AsEnumerable):
+            case nameof(Enumerable.Cast):
+            case "Chunk":
+            case nameof(Enumerable.Concat):
+            case nameof(Enumerable.Distinct):
+            case "DistinctBy":
+            case nameof(Enumerable.Empty):
+            case nameof(Enumerable.Except):
+            case nameof(Enumerable.GroupBy):
+            case nameof(Enumerable.GroupJoin):
+            case nameof(Enumerable.Intersect):
+            case "IntersectBy":
+            case nameof(Enumerable.Join):
+            case nameof(Enumerable.OfType):
+            case nameof(Enumerable.OrderBy):
+            case nameof(Enumerable.OrderByDescending):
+            case "Prepend":
+            case nameof(Enumerable.Repeat):
+            case nameof(Enumerable.Reverse):
+            case nameof(Enumerable.Select):
+            case nameof(Enumerable.SelectMany):
+            case nameof(Enumerable.Skip):
+            case nameof(Enumerable.SkipWhile):
+            case nameof(Enumerable.Take):
+            case nameof(Enumerable.TakeWhile):
+            case nameof(Enumerable.ThenBy):
+            case nameof(Enumerable.ThenByDescending):
+            case nameof(Enumerable.ToArray):
+            case nameof(Enumerable.ToDictionary):
+            case nameof(Enumerable.ToList):
```

Comment:
ToHashSet

Reply:
Preview

Reply:
> Preview
Why do you think so? It is even in .Net Framework 4.8:

![image](https://user-images.githubusercontent.com/103252490/192463559-d8f76b0c-7741-41e3-a46f-b0cb73af8332.png)

https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.tohashset?view=net-6.0

Reply:
Ah, yes.